proposed/activity-improvements.md

```diff
@@ -0,0 +1,290 @@
+```
+
+
+## Proposed API
+
+### ActivityKind
+
+```C#
+namespace System.Diagnostics
+{
+    // Kind describes the relationship between the Activity, its parents, and its children in a Trace.
+    public enum ActivityKind
+    {
+        // Indicates that the Activity represents an internal operation within an application, as opposed to an operation with remote parents or children.
+        Internal = 1,
+
+        /// Server activity represents request incoming from external component.
+        Server = 2,
+
+        /// Client activity represents outgoing request to the external component.
+        Client = 3,
+
+        /// Producer activity represents output provided to external components.
+        Producer = 4,
+
+        /// Consumer activity represents output received from an external component.
+        Consumer = 5,
+    }
+}
+```
+
+ ### ActivityContext
+
+```C#
+    /// ActivityContext representation conforms to the w3c TraceContext specification. It contains two identifiers
+    /// a TraceId and a SpanId - along with a set of common TraceFlags and system-specific TraceState values.
+    public readonly struct ActivityContext : IEquatable<ActivityContext>
+    {
+        public ActivityContext(ActivityTraceId traceId, ActivitySpanId spanId, ActivityTraceFlags traceFlags, string? traceState = null)
+        public ActivityTraceId TraceId { get; }
+        public ActivitySpanId SpanId { get; }
+        public ActivityTraceFlags TraceFlags { get; }
+        public string? TraceState { get; }
+
+        public static bool operator ==(ActivityContext context1, ActivityContext context2)
+        public static bool operator !=(ActivityContext context1, ActivityContext context2)
+        public bool Equals(ActivityContext context)
+        public override bool Equals(object? obj)
+        public override int GetHashCode()
+    }
+```
+
+ ### ActivityLink
+```C#
+namespace System.Diagnostics
+{
+    /// <summary>
+    /// Activity may be linked to zero or more other <see cref="ActivityContext"/> that are causally related.
+    /// Links can point to ActivityContexts inside a single Trace or across different Traces.
+    /// Links can be used to represent batched operations where a Activity was initiated by multiple initiating Activities,
+    /// each representing a single incoming item being processed in the batch.
+    public readonly struct ActivityLink
+    {
+        public ActivityLink(ActivityContext context) 
+        public ActivityLink(ActivityContext context, IDictionary<string, object>? attributes)
+        public ActivityContext Context { get; }
+        public IDictionary<string, object>? Attributes { get; }
+    }
+}
+```
+
+### Activity
+
+```C#
+namespace System.Diagnostics
+{
+    public partial class Activity : IDisposable // added IDisposable 
```

Comment:
Moving discussion from [the old issue](https://github.com/dotnet/runtime/issues/31373#issuecomment-600908953) to here.  

@SergeyKanzhelev wrote

> UpdateName is absolutely needed. Asp.Net Core listener wouldn't work without it: https://github.com/open-telemetry/opentelemetry-dotnet/blob/633dda37bdb171f337596ba4b53021b5d35f117c/src/OpenTelemetry.Collector.AspNetCore/Implementation/HttpInListener.cs#L148

Reply:
What if we have Activity.DisplayName property? Default value of the property is set to OperationName but the setter can be used to change it at any time.

I think adding a setter to Activity.OperationName is very likely to be breaking for pre-existing telemetry so if not DisplayName we should brainstorm something else.